src/button_input/controller.py

```diff
@@ -0,0 +1,69 @@
+import threading
+from inputs import get_gamepad
+from enum import Enum
+
+class ControllerButton(Enum):
+    LeftJoystickY = 1
+    LeftJoystickX = 2
+    RightJoystickY = 3
+    RightJoystickX = 4
+    LeftTrigger = 5
+    RightTrigger = 6
+    LeftBumper = 7
+    RightBumper = 8
+    A = 9
+    X = 10
+    Y = 11
+    B = 12
+    LeftThumb = 13
+    RightThumb = 14
+    Back = 15
+    Start = 16
+    LeftDPad = 17
+    RightDPad = 18
+    UpDPad = 19
+    DownDPad = 20
+
+class XboxController:
+    def __init__(self, add_controller_note):
+        # Map of event codes to controller buttons
+        self.event_button_map = {
+            'ABS_Y': ControllerButton.LeftJoystickY,
+            'ABS_X': ControllerButton.LeftJoystickX,
+            'ABS_RY': ControllerButton.RightJoystickY,
+            'ABS_RX': ControllerButton.RightJoystickX,
+            'ABS_Z': ControllerButton.LeftTrigger,
+            'ABS_RZ': ControllerButton.RightTrigger,
+            'BTN_TL': ControllerButton.LeftBumper,
+            'BTN_TR': ControllerButton.RightBumper,
+            'BTN_SOUTH': ControllerButton.A,
+            'BTN_EAST': ControllerButton.B,
+            'BTN_NORTH': ControllerButton.Y,
+            'BTN_WEST': ControllerButton.X,
+            'BTN_THUMBL': ControllerButton.LeftThumb,
+            'BTN_THUMBR': ControllerButton.RightThumb,
+            'BTN_SELECT': ControllerButton.Back,
+            'BTN_START': ControllerButton.Start,
+            'BTN_DPAD_LEFT': ControllerButton.LeftDPad,
+            'BTN_DPAD_RIGHT': ControllerButton.RightDPad,
+            'BTN_DPAD_UP': ControllerButton.UpDPad,
+            'BTN_DPAD_DOWN': ControllerButton.DownDPad
+        }
+
+        self.add_controller_note = add_controller_note
```

Comment:
I don't like the name of this callback much, it doesn't convey immediately that this is a callback. Just a nitpick tho.